core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -0,0 +1,194 @@
+    Map<String, BlockMetaInfo> fileNameToMetaInfoMapping = new TreeMap();
+    CarbonFile carbonFile = FileFactory.getCarbonFile(segmentFilePath);
+    if (carbonFile instanceof AbstractDFSCarbonFile) {
+      Path path = new Path(segmentFilePath);
+      RemoteIterator<LocatedFileStatus> iter =
+          ((AbstractDFSCarbonFile) carbonFile).fs.listLocatedStatus(path);
+      PathFilter pathFilter = new PathFilter() {
+        @Override public boolean accept(Path path) {
+          return CarbonTablePath.isCarbonDataFile(path.getName());
+        }
+      };
+      while (iter.hasNext()) {
+        LocatedFileStatus fileStatus = iter.next();
+        if (pathFilter.accept(fileStatus.getPath())) {
+          String[] location = fileStatus.getBlockLocations()[0].getHosts();
+          long len = fileStatus.getLen();
+          BlockMetaInfo blockMetaInfo = new BlockMetaInfo(location, len);
+          fileNameToMetaInfoMapping.put(fileStatus.getPath().toString(), blockMetaInfo);
+        }
+      }
+    }
+    return fileNameToMetaInfoMapping;
+  }
+
+  private static BlockMetaInfo createBlockMetaInfo(
+      Map<String, BlockMetaInfo> fileNameToMetaInfoMapping, String carbonDataFile)
+      throws IOException {
+    FileFactory.FileType fileType = FileFactory.getFileType(carbonDataFile);
+    switch (fileType) {
+      case LOCAL:
+        CarbonFile carbonFile = FileFactory.getCarbonFile(carbonDataFile, fileType);
+        return new BlockMetaInfo(new String[] { "localhost" }, carbonFile.getSize());
+      default:
+        return fileNameToMetaInfoMapping.get(carbonDataFile);
+    }
+  }
+
+  public static Set<TableBlockIndexUniqueIdentifier> getTableBlockUniqueIdentifiers(Segment segment)
+      throws IOException {
+    Set<TableBlockIndexUniqueIdentifier> tableBlockIndexUniqueIdentifiers = new HashSet<>();
+    Map<String, String> indexFiles = segment.getCommittedIndexFile();
+    for (Map.Entry<String, String> indexFileEntry : indexFiles.entrySet()) {
+      Path indexFile = new Path(indexFileEntry.getKey());
+      tableBlockIndexUniqueIdentifiers.add(
+          new TableBlockIndexUniqueIdentifier(indexFile.getParent().toString(), indexFile.getName(),
+              indexFileEntry.getValue(), segment.getSegmentNo()));
+    }
+    return tableBlockIndexUniqueIdentifiers;
+  }
+
+  /**
+   * This method will filter out the TableBlockIndexUniqueIdentifier belongs to that distributable
+   *
+   * @param tableBlockIndexUniqueIdentifiers
+   * @param distributable
+   * @return
+   */
+  public static TableBlockIndexUniqueIdentifier filterIdentifiersBasedOnDistributable(
+      Set<TableBlockIndexUniqueIdentifier> tableBlockIndexUniqueIdentifiers,
+      BlockletDataMapDistributable distributable) {
+    TableBlockIndexUniqueIdentifier validIdentifier = null;
+    for (TableBlockIndexUniqueIdentifier tableBlockIndexUniqueIdentifier :
+        tableBlockIndexUniqueIdentifiers) {
+      if (distributable.getFilePath().equals(
```

Comment:
Better get the file name from distributable and compare only file name instead of comparing complete path